### PR TITLE
应该是非空才循环，不然会导致在使用redis注册中心时消费者引用不到服务

### PR DESCRIPTION
**Pull request by boomblog**
## What is the purpose of the change

在使用redis做为注册中心时，发现服务不能引用

## Brief changelog
CollectionUtils.isEmptyMap(values)改为if (!CollectionUtils.isEmptyMap(values)) 

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by CrazyHZM** [comment]
Looks good to me.

File: dubbo-registry/dubbo-registry-redis/src/main/java/org/apache/dubbo/registry/redis/RedisRegistry.java [hunk 1]
```diff
@@ -428,7 +428,7 @@ private void doNotify(Jedis jedis, Collection<String> keys, URL url, Collection<
             }
             List<URL> urls = new ArrayList<>();
             Map<String, String> values = jedis.hgetAll(key);
-            if (CollectionUtils.isEmptyMap(values)) {
+            if (!CollectionUtils.isEmptyMap(values)) {
```
Comment (khanimteyaz) [inline dubbo-registry/dubbo-registry-redis/src/main/java/org/apache/dubbo/registry/redis/RedisRegistry.java]: Thanks for this PR. Could you add a UT for this?

Comment (boomblog) [reply dubbo-registry/dubbo-registry-redis/src/main/java/org/apache/dubbo/registry/redis/RedisRegistry.java]: Is it necessary? This is my first PR，I don't know much about。

Comment (khanimteyaz) [reply dubbo-registry/dubbo-registry-redis/src/main/java/org/apache/dubbo/registry/redis/RedisRegistry.java]: @boomblog no worries I will add that on behalf of you 😃 . As a practice we follow to add UT if there is conditional changes. Do us a favour create an issue with information like UT and provide a reference of this PR.

Thanks for raising this PR and the contribution. Welcome.

[the remark this answers is not in the document]
Comment (CrazyHZM) [reply dubbo-registry/dubbo-registry-redis/src/main/java/org/apache/dubbo/registry/redis/RedisRegistry.java]: @boomblog This suggestion is good, please modify it.

**Pull request merged**